### PR TITLE
Fix: Revert opcode

### PR DESCRIPTION
**Pull request by ArthurBraud**
Fix Revert

### CODE REVIEW

**Review by assimovt — APPROVED** [review]

File: docs/opcodes/FD.mdx [hunk 1]
```diff
@@ -5,18 +5,30 @@ group: System operations
 
 ## Notes
 
-Equivalent to [CREATE](/#F0), except the salt allows to create at a deterministic address.
+Abort execution, undo all state changes, return a value and refund the remaining gas to the caller.
```
Comment (peroket) [inline docs/opcodes/FD.mdx:8]: Thanks, but this is not complete :-)
FD should be renamed to F5 first, and then FD can take those contents.

File: docs/opcodes/FD.mdx [hunk 2]
```diff
@@ -5,18 +5,30 @@ group: System operations
...
+The error message returned is available to the caller in the returndata buffer.
```
Comment (peroket) [inline docs/opcodes/FD.mdx:9]: Not "error message", but rather "The [return data](/about) of the calling context is set as the given chunk of memory of this context."

File: docs/opcodes/FD.mdx (hunk 1 above)
Comment (peroket) [reply docs/opcodes/FD.mdx:8]: That's nice but not complete :-)
I would say "Stop the current context execution, revert the state changes (see [STATICCALL](/#FA) for a list of state changing opcodes) and return unused gas to the caller. If the execution is stopped with REVERT, the value `0` is put on the stack of the calling context, which continues to execute normally".

"Refund" has a specific meaning, let's not use it here, it's not the same thing.

File: docs/opcodes/FD.mdx [hunk 3]
```diff
@@ -5,18 +5,30 @@ group: System operations
...
 
 ## Stack input
 
-0. `Integer`: value in [wei](https://www.investopedia.com/terms/w/wei.asp) to send to the new account.
-1. `Integer`: offset in the [memory](/about) in bytes, the instructions of the new account.
-2. `Integer`: how many bytes to copy (size of the instructions).
-3. `Integer`: salt, 32-byte value used to create the new account at a deterministic address.
+0. `Integer`: offset in the [memory](/about) in bytes. The starting position of the value to return.
```
Comment (peroket) [inline docs/opcodes/FD.mdx:13]: The starting position of the value to return -> The return data of the calling context

File: docs/opcodes/FD.mdx [hunk 4]
```diff
@@ -5,18 +5,30 @@ group: System operations
...
+1. `Integer`: length in the [memory](/about) in bytes to return.
```
Comment (peroket) [inline docs/opcodes/FD.mdx:14]: For consistency: "how many bytes to copy (size of the [return data](/about))."

File: docs/opcodes/FD.mdx [hunk 5]
```diff
@@ -5,18 +5,30 @@ group: System operations
...
+
+## Returndata output
```
Comment (peroket) [inline docs/opcodes/FD.mdx:16]: This is not consistent, please remove that title and its content

File: docs/opcodes/FD.mdx [hunk 6]
```diff
@@ -5,18 +5,30 @@ group: System operations
...
+
+memory[offset:offset+length]
+
+## Examples
+
+| Memory |
```
Comment (peroket) [inline docs/opcodes/FD.mdx:22]: Here you can add another array (at the bottom) to show the actual bytes of the return data according to the given example memory and parameters. You can call it "Calling context return data"

**Review by peroket — CHANGES_REQUESTED** [review]
Thanks for starting it for me :-)

File: docs/opcodes/FD.mdx (hunk 1 above)
Comment (ArthurBraud) [reply docs/opcodes/FD.mdx:8]: Sounds good, I'll change it

File: docs/opcodes/FD.mdx (hunk 2 above)
Comment (ArthurBraud) [reply docs/opcodes/FD.mdx:9]: Change it

File: docs/opcodes/FD.mdx (hunk 3 above)
Comment (ArthurBraud) [reply docs/opcodes/FD.mdx:13]: +1 changed it

**Comment by ArthurBraud** [comment]
ptal @peroket

File: docs/opcodes/FD.mdx [hunk 7]
```diff
@@ -5,18 +5,30 @@ group: System operations
...
-2. `Integer`: how many bytes to copy (size of the instructions).
-3. `Integer`: salt, 32-byte value used to create the new account at a deterministic address.
+0. `Integer`: Offset in the [memory](/about) in bytes. The return data of the calling context.
+1. `Integer`: Length: how many bytes to copy (size of the [return data](/about).
+
+## Examples
+
+| Memory |
+|--------|
+| `0xFF01000000000000000000000000000000000000000000000000000000000000` |
```
Comment (peroket) [inline docs/opcodes/FD.mdx:20]: If the zeros are not useful in your example, please remove them, to make it more clear

File: docs/opcodes/FD.mdx [hunk 8]
```diff
@@ -5,18 +5,30 @@ group: System operations
...
+
+|  Stack Input |
+|-------------:|
+|   `0x0001`   |
```
Comment (peroket) [inline docs/opcodes/FD.mdx:24]: I suppose you want to represent two values, but you did only one with extra zeros.
It should be like that:
```
| Input |
|------:|
| `0` |
| `2` |
```
You're copying two bytes in your example, so the parameter length should be 2.
Also note the title and no extra space for consistency with the rest :-)

File: docs/opcodes/FD.mdx [hunk 9]
```diff
@@ -5,18 +5,30 @@ group: System operations
...
+
+| Calling context return data |
+|----------------------------:|
+|           `0xFF01`          |
```
Comment (peroket) [inline docs/opcodes/FD.mdx:28]: Please remove the extra spaces for consistency

File: docs/opcodes/FD.mdx [hunk 10]
```diff
@@ -5,18 +5,30 @@ group: System operations
...
+Stop the current context execution, revert the state changes (see [STATICCALL](/#FA) for a list of state changing opcodes) and return unused gas to the caller. If the execution is stopped with REVERT, the value 0 is put on the stack of the calling context, which continues to execute normally.
+The [return data](/about) of the calling context is set as the given chunk of memory of this context.
 
 ## Stack input
 
-0. `Integer`: value in [wei](https://www.investopedia.com/terms/w/wei.asp) to send to the new account.
-1. `Integer`: offset in the [memory](/about) in bytes, the instructions of the new account.
-2. `Integer`: how many bytes to copy (size of the instructions).
-3. `Integer`: salt, 32-byte value used to create the new account at a deterministic address.
+0. `Integer`: Offset in the [memory](/about) in bytes. The return data of the calling context.
```
Comment (peroket) [inline docs/opcodes/FD.mdx:13]: Please remove capital letter for consistency: Offset -> offset

File: docs/opcodes/FD.mdx [hunk 11]
```diff
@@ -5,18 +5,30 @@ group: System operations
...
+1. `Integer`: Length: how many bytes to copy (size of the [return data](/about).
```
Comment (peroket) [inline docs/opcodes/FD.mdx:14]: Please remove "Length: " for consistency

**Review by peroket — CHANGES_REQUESTED** [review]

**Comment by peroket** [comment]
> ptal @peroket

FYI you don't need to add a comment for that, just use github "request a review" feature :-)

File: docs/opcodes/FD.mdx (hunk 11 above)
Comment (peroket) [reply docs/opcodes/FD.mdx]: This has not been done :-)

Comment (ArthurBraud) [reply docs/opcodes/FD.mdx]: I removed it

**Review by peroket — APPROVED** [review]

**Pull request merged**